### ISSUE TITLE
Regression in 0.1.0: dynamic removal of styles from shadow host not handled in Safari

### ISSUE THREAD

**Issue by web-padawan** [LeaVerou/style-observer#120]
Consider the following example, it worked with 0.0.9 but doesn't work after upgrading to 0.1.0 in Safari 18.5
Steps to reproduce: click "Add styles", then "Remove styles" - there should be two warnings in the console.

When adding styles to the document instead using `document.head.appendChild(style)`, it works fine.

```html
<div id="host"></div>

<p>
  <button id="addStyle">Add styles</button>
  <button id="removeStyle">Remove styles</button>
</p>

<script type="module">
  import { StyleObserver } from 'style-observer';

  const host = document.querySelector('#host');
  host.attachShadow({ mode: 'open' });
  host.shadowRoot.innerHTML = 'Content';

  // Observe host style property change
  CSS.registerProperty({
    name: '--test-foo-css-inject',
    syntax: '<number>',
    inherits: true,
    initialValue: '0',
  });

  const observer = new StyleObserver((records) => {
    records.forEach((record) => {
      const { property, value, oldValue } = record;
      if (property === '--test-foo-css-inject') {
        console.warn('value', value, 'old', oldValue); // Should report 1 -> 0 when removing style
      }
    });
  });

  observer.observe(host, '--test-foo-css-inject');

  // Dynamically add and remove <style> tag
  let style;

  document.querySelector('#addStyle').addEventListener('click', () => {
    if (!style) {
      style = document.createElement('style');
      style.textContent = `
        html, :host {
          --test-foo-css-inject: 1;
        }
      `;
    }
     
    host.shadowRoot.appendChild(style); // Does not work
  });

  document.querySelector('#removeStyle').addEventListener('click', () => {
    if (style) {
      style.remove();
    }
  });
</script>
```

**Comment by LeaVerou** [comment]
Still investigating. Some finds:
- Logging in `handleEvent()`. No transition event gets fired when the `<style>` is removed.
- Hardcoding any combination of values for both detected bugs. Nothing.
- Changing the initial value to 1 and the CSS set value to 0. Same thing, now it fires for 1 -> 0 but not when the style gets removed.
- Setting the transition properties we set with `!important`. No difference.
- Instead of removing the `<style>`, setting 0 as an inline style. **Transition fires!**
- Changing the CSS rule to set to 0 instead of removing the style (`host.style.setProperty("--test-foo-css-inject", "0")`). **Transition fires!**
- Changing the CSS rule to set to `initial` instead of removing the style (`host.style.setProperty("--test-foo-css-inject", "initial")`). No difference.
- Removing the declaration (`style.sheet.cssRules[0].style.removeProperty("--test-foo-css-inject")`. No difference.
- Deleting the CSS rule (`style.sheet.deleteRule(0)`). No difference.

So it's very clearly a Safari bug if `initial` has a different behavior than `0`. I think the best way forwards is to detect the bug and fall back to the old way in that case (just like we do when `adoptedStylesheets` are not available, so the code path is already there).

**Comment by LeaVerou** [comment]
@web-padawan sorry for the delay, can you please verify that this is now fixed?
